synctl

```diff
@@ -228,6 +238,7 @@ def main():
     start_stop_synapse = True
 
     if cache_factor:
+        logger.warning(SYNCTL_CACHE_FACTOR_WARNING)
         os.environ["SYNAPSE_CACHE_FACTOR"] = str(cache_factor)
```

Comment:
If we are warning and stating that this option is deprecated, should we still set the env variable on the next line?

Reply:
I think so. Doesn't deprecating a feature just _prepare_ the user that the feature will be removed in the feature? Otherwise it would be a feature removal, not a deprecation. However, the [above mentioned comment](https://github.com/matrix-org/synapse/issues/6354#issuecomment-552811915) demands a deprecation, not a removal.  
But of course, I can also remove it.